public/src/managers/ServerProvider.ts

```diff
@@ -455,4 +451,8 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
         };
         return isbm;
     }
+
+    private responseCodeSuccess(response: IGrpcResponse<any>): boolean {
+        return response.status.getCode() === 0;
+    }
```

Comment:
This can also check the `response.data` field as part of the check. That way, we can skip many of the checks above.
```typescript
    return response.status.getCode() === 0 && result.data;
```
(maybe the method name needs to be changed too, but that's fine.)

Reply:
Unfortunately, it cannot (this was my first approach as well). The calling method will still refuse to believe that `response.data` is present if checked like this, so it won't compile. But I agree, there has to be a more elegant solution.

Reply:
Honestly, it looks like making sure that both status is `0` and `data` field is present is pretty redundant: there will be no data if the status is not 0, and there could not be any other status code then 0 if the data is here.

The only reason we check now is because the `data` field is optional and TS with its strictness will not allow working with it unless we check explicitly that the field is set.

It needs some testing, but for now it looks reasonable to change the `data` field from optional to required, and only look at the status code.

Reply:
All fields in proto3 are required.

Reply:
OK, making the field required does not really look like a good idea because of the generic.

Reply:
`IGrpcResponse<T>` is not a Protobuf type. The `data` field in question belongs to it.